testcontainers/redis-cluster-compose.yml

```diff
@@ -0,0 +1,17 @@
+version: '3.3'
+services:
+  redis-cluster:
+    image: 'redis:6.2.5-alpine'
+    container_name: redis
+    volumes:
+      - /home/alsergeev/workspace/zio/zio-redis/testcontainers/redis-cluster.sh:/data/redis-cluster.sh
```

Comment:
I don't think this is going to work :). Also, you may use something like this without the need to split the single compose file, and introduce custom scripts:

```yaml
  redis-node-0:
    image: docker.io/bitnami/redis-cluster:6.2
    environment:
      - 'ALLOW_EMPTY_PASSWORD=yes'
      - 'REDIS_NODES=redis-node-0 redis-node-1 redis-node-2 redis-node-3 redis-node-4 redis-node-5'

  redis-node-1:
    image: docker.io/bitnami/redis-cluster:6.2
    environment:
      - 'ALLOW_EMPTY_PASSWORD=yes'
      - 'REDIS_NODES=redis-node-0 redis-node-1 redis-node-2 redis-node-3 redis-node-4 redis-node-5'

  redis-node-2:
    image: docker.io/bitnami/redis-cluster:6.2
    environment:
      - 'ALLOW_EMPTY_PASSWORD=yes'
      - 'REDIS_NODES=redis-node-0 redis-node-1 redis-node-2 redis-node-3 redis-node-4 redis-node-5'

  redis-node-3:
    image: docker.io/bitnami/redis-cluster:6.2
    environment:
      - 'ALLOW_EMPTY_PASSWORD=yes'
      - 'REDIS_NODES=redis-node-0 redis-node-1 redis-node-2 redis-node-3 redis-node-4 redis-node-5'

  redis-node-4:
    image: docker.io/bitnami/redis-cluster:6.2
    environment:
      - 'ALLOW_EMPTY_PASSWORD=yes'
      - 'REDIS_NODES=redis-node-0 redis-node-1 redis-node-2 redis-node-3 redis-node-4 redis-node-5'

  redis-node-5:
    image: docker.io/bitnami/redis-cluster:6.2
    depends_on:
      - redis-node-0
      - redis-node-1
      - redis-node-2
      - redis-node-3
      - redis-node-4
    environment:
      - 'ALLOW_EMPTY_PASSWORD=yes'
      - 'REDIS_CLUSTER_REPLICAS=1'
      - 'REDIS_NODES=redis-node-0 redis-node-1 redis-node-2 redis-node-3 redis-node-4 redis-node-5'
      - 'REDIS_CLUSTER_CREATOR=yes'
    ports:
      - "6379:6379"
 ```

Reply:
Ok. I spent half of a day trying your suggestion. It doesn't work or i don't know how to do it work(( I can't change ports in cluster to expose them, for redis cluster ports that expose have to equals node port. I suppose bitnami has a bug in scripts that starting cluster.
My script is simple from redis documentation, i suggest stick to it.

Reply:
I literally copied this setup from a working test setup I have, so, yes, it works.

Reply:
You can leave the script, but please don't restructure the directories.

Reply:
Have u run tests? What works?

Reply:
Let's go one more time: the script i pasted is copied from a working test setup I have and I use at work, so yes, I am running tests against it :).

Nevertheless, to end this discussion as it is already going the wrong way, as I said in the comment above: please keep the script you added and don't restructure the compose files .

Reply:
Sorry, if u understand me wrong. I'm just trying to figure out, what i'm doing wrong, cause for me it doesn't work.
It's starting cluster - yes. But i can't connect to it cause our tests not in the same network that this containers. That's why i have to expose ports, but if exposing ports do not equal redis node ports, redis cluster config know nothing about expose ports and our client would try to connect to redis node ports inside container.

Reply:
OK. Yeah my local filesystem references i already deleted)

Reply:
> You can leave the script, but please don't restructure the directories.

Are you asking to delete testcontainers folder?

Reply:
Yes.

Reply:
Isn't it better to have every thing tidy and not in one place?
I like everything in it's folders and with logic

Reply:
Ok.

Reply:
> OK. Yeah my local filesystem references i already deleted) 

I presume you didn't push that yet cause i still see it :)

Reply:
Interesting. May this is because im force pushed some commits 🤔